src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -1104,7 +1104,14 @@ private SamlSecurityToken ValidateSignature(SamlSecurityToken samlToken, string
                 if (keyMatched)
                     throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidSignatureException(LogHelper.FormatInvariant(TokenLogMessages.IDX10514, keysAttempted, samlToken.Assertion.Signature.KeyInfo, exceptionStrings, samlToken)));
 
-                throw LogHelper.LogExceptionMessage(new SecurityTokenSignatureKeyNotFoundException(LogHelper.FormatInvariant(TokenLogMessages.IDX10513, samlToken.Assertion.Signature.KeyInfo, exceptionStrings, samlToken)));
+                if (samlToken.Assertion.Conditions != null && samlToken.Assertion.Signature.KeyInfo != null)
```

Comment:
samlToken.Assertion.Signature.KeyInfo should be already verified at this point.

Could we do something like:
samlToken.Assertion.Conditions?.NotBefore
samlToken.Assertion.Conditions?.NotOnOrAfter

and call into InternalValidators?

Reply:
Will remove the check on Signature.KeyInfo.

Saml1 doesn't use nullable datetimes, so can only check the conditions object for null.

For saml2, I think we pass along null NotBefore and NotOnOrAfter and let that contribute to the decision on lifetime validity. Does that sound right?